tools/mo/openvino/tools/mo/front/tf/MapFNTransformation.py

```diff
@@ -255,3 +300,97 @@ def find_and_replace_pattern(self, graph: Graph):
                                               internal_match['increment_iteration_result'].internal_layer_id,
                                               internal_match['current_iteration'].internal_layer_id):
                     MapFNOutputConcatenation.transform_map_fn_output_concatenation(external_match, internal_match)
+
+
+class TensorListOutputConcatenation(FrontReplacementSubgraph):
+    """
+    The transformation handles inputs slicing in While loop. It avoids TensorListPushBack, and EmptyTensorList
+    operations and replaces the original sub-graph by adding axis attribute in Loop node for concatenation of
+    intermediate output results.
+    """
+    enabled = True
+
+    def run_before(self):
+        return [WhileNormalize]
+
+    @staticmethod
+    def get_body_pattern():
+        return dict(
+            nodes=[('container', dict(op='Parameter')),
+                   ('concatenation', dict(op='TensorListPushBack')),
+                   ('concatenation_result', dict(op='Result'))
+                   ],
+            edges=[
+                   ('container', 'concatenation', {'in': 0}),
+                   ('concatenation', 'concatenation_result', {'in': 0}),
+                   ]
+        )
+
+    @staticmethod
+    def transform_tensor_list_output_concatenation(external_match: dict, internal_match: dict):
+        """
+        Transforms TensorFlow 2 output concatenation into use of axis attribute for output port of Loop node
+        :param external_match: a match used for handling a part of the main graph responsible for output concatenation
+        :param internal_match: a match used for handling a part of the body graph responsible for output concatenation
+        """
+        loop_node = external_match['while']
```

Comment:
Big part of this function duplicated from transform_map_fn_output_concatenation. Could we create separate function for this code and use it in both places?

Reply:
Unfortunately they are not fully duplicates and if we create a separate function it will be hard understandable and not clear. So I prefer to have 2 functions.`